tests/integration/metadata/test_metadata.py

```diff
@@ -274,3 +278,82 @@ def test_single_table_compatibility(tmp_path):
+    # Setup
+    data, _ = download_demo('multi_table', 'fake_hotels')
+    warn_msg = re.escape(
+        "The 'MultiTableMetadata' is deprecated. Please use the new "
+        "'Metadata' class for synthesizers."
+    )
+
+    multi_dict = {
+        'tables': {
+            'guests': {
+                'primary_key': 'guest_email',
+                'columns': {
+                    'guest_email': {'sdtype': 'email', 'pii': True},
+                    'hotel_id': {'sdtype': 'id', 'regex_format': '[A-Za-z]{5}'},
+                    'has_rewards': {'sdtype': 'boolean'},
+                    'room_type': {'sdtype': 'categorical'},
+                    'amenities_fee': {'sdtype': 'numerical', 'computer_representation': 'Float'},
+                    'checkin_date': {'sdtype': 'datetime', 'datetime_format': '%d %b %Y'},
+                    'checkout_date': {'sdtype': 'datetime', 'datetime_format': '%d %b %Y'},
+                    'room_rate': {'sdtype': 'numerical', 'computer_representation': 'Float'},
+                    'billing_address': {'sdtype': 'address', 'pii': True},
+                    'credit_card_number': {'sdtype': 'credit_card_number', 'pii': True},
+                },
+            },
+            'hotels': {
+                'primary_key': 'hotel_id',
+                'columns': {
+                    'hotel_id': {'sdtype': 'id', 'regex_format': 'HID_[0-9]{3}'},
+                    'city': {'sdtype': 'categorical'},
+                    'state': {'sdtype': 'categorical'},
+                    'rating': {'sdtype': 'numerical', 'computer_representation': 'Float'},
+                    'classification': {'sdtype': 'categorical'},
+                },
+            },
+        },
+        'relationships': [
+            {
+                'parent_table_name': 'hotels',
+                'parent_primary_key': 'hotel_id',
+                'child_table_name': 'guests',
+                'child_foreign_key': 'hotel_id',
+            }
+        ],
+        'METADATA_SPEC_VERSION': 'MULTI_TABLE_V1',
+    }
+    metadata = MultiTableMetadata.load_from_dict(multi_dict)
+    assert type(metadata) is MultiTableMetadata
+
+    # Run
+    with pytest.warns(FutureWarning, match=warn_msg):
+        synthesizer = HMASynthesizer(metadata)
+    synthesizer.fit(data)
+    model_path = os.path.join(tmp_path, 'synthesizer.pkl')
+    synthesizer.save(model_path)
+
+    # Assert
+    assert os.path.exists(model_path)
+    assert os.path.isfile(model_path)
+    loaded_synthesizer = HMASynthesizer.load(model_path)
+    assert isinstance(synthesizer, HMASynthesizer)
+    assert loaded_synthesizer.get_info() == synthesizer.get_info()
+    assert isinstance(loaded_synthesizer.metadata, Metadata)
+    expected_metadata = metadata.to_dict()
+    expected_metadata['METADATA_SPEC_VERSION'] = 'V1'
+    assert loaded_synthesizer.metadata.to_dict() == expected_metadata
+    loaded_sample = loaded_synthesizer.sample(10)
+    synthesizer.validate(loaded_sample)
```

Comment:
Could we split this into multiple sections:
```
# Load HMASynthesizer
# Asserts
# Sample from loaded synthesizer
```

Reply:
Fixed